manuscripts2/Readme.md

```diff
@@ -0,0 +1,230 @@
+# Readme
+
+This folder is an initialization to the new functions and classes that have to be added to manuscripts. This file provides a brief introduction to these classes and functions.
+
+
+### TODO:
+- Is popping from dict and then creating the nested aggregation the most efficient way to do this?
+- How do you parse 3 level nested aggregation? 
+
+
+### Structure:
+The main Class which is being implemented is the `EQCC` class. `EQCC`  stands for `Elasticsearch Query, Connection and Compute` as it provides a connection to elasticsearch, queries it and computes the response.
+
+The important variables inside the `EQCC` objects are as follows:
+
+- `s`: This is the elasticsearch_dsl Search object which is responsible of quering elasticsearch and getting the results. All the aggregations, queries and filters are applied to this Search object and then it queries elasticsearch and fetches the required results.
+- `parent_id`: This is a counter counting the aggregations that are applied to the Search object. It starts with `0` and increments as aggregations are added.
+- `queries`: This is a dictionary containing two lists: `must` and `must_not`. This dictionary contains the normal(`must`) and inverse(`must_not`) queries.
+- `aggregations`: This is an OrderedDict which contains all the aggregations applied. An ordered dict allows us to create nested aggregations easily, as we'll see below.
```

Comment:
@jgbarah The way this class is implemented, I need a data structure which stores the data in an ordered manner so that nested aggregations can be created by chaining two aggregations (or applying two aggregations one after the other: the second one being one of `by_authors`, `by_organizations`, `by_period`) and first one being the child aggregation of the second.
```
issues = Issues(index)
issues.get_cardinality("id_in_repo").by_period("author_name")
``` 
In the first step, the first aggregation (`get_cardinality`) will be added to the dict and on applying the second aggregation(`by_period`), the first aggregation will be `popped` from the dict and added as a child aggregation to the second agg.